src/client/app/components/visual-group/CreateVisualGroupComponent.tsx

```diff
@@ -0,0 +1,851 @@
+
+
+import * as React from 'react';
+import * as d3 from 'd3';
+import { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useAppSelector } from '../../redux/reduxHooks';
+import { GroupData } from '../../../../client/app/types/redux/groups'
+import { MeterData } from '../../../../client/app/types/redux/meters'
+import { selectAllMeters } from '../../redux/api/metersApi';
+import { selectAllGroups, groupsApi } from '../../redux/api/groupsApi';
+
+/**
+ *   Visual graph component that shows the relationship between all groups and meters
+ * entered by an admin
+ * @returns D3 force graph visual
+ */
+interface CreateVisualGroupProps {
+    groups: GroupData[];
+    meters: MeterData[];
+}
+
+//Declare node types for groups and meters, will determine every node's design schema on the graphic
+type GroupNodeType = 'unselectedGroup' | 'selectedGroup' | 'childGroup' | 'deepGroup';
+type MeterNodeType = 'meter' | 'childMeter' | 'deepMeter';
+type AllNodeType = GroupNodeType | MeterNodeType;
+
+export const CreateVisualGroupComponent: React.FC<CreateVisualGroupProps> = ({
+    groups,
+    meters
+}) => {
+
+    const [selectedGroupId, setSelectedGroupId] = useState<number | null>(null);
+
+    const intl = useIntl();
+
+    /*Get Meter data from redux*/
+    const allMeters: MeterData[] = useAppSelector(selectAllMeters);
+    
+    /* Fetch deep groups data from API */
+    const { data: deepGroupsData = [], isLoading: deepGroupsLoading } = groupsApi.useGetAllGroupsDeepGroupsQuery();
+
+    /* Get all Group data from Redux */
+    const allGroups: GroupData[] = useAppSelector(selectAllGroups);
+    
+    /* Merge deepGroups data with allGroups */
+    const mergedGroups: GroupData[] = allGroups.map(group => {
+        const deepGroupData = deepGroupsData.find(dg => dg.id === group.id);
+        return {
+            ...group,
+            deepGroups: deepGroupData?.deepGroups || []
+        };
+    });
+
+
+    const selectedGroup: GroupData | undefined = mergedGroups.find(group => group.id === selectedGroupId);
+
+    /*Get all meters who are in any way a children of a group */
+    const groupedMeterIds: Set<number> = new Set(
+        mergedGroups.flatMap(group => group.deepMeters)
+    );
+
+    /*Only keep meters who have a relationship with a group */
```

Comment:
The comments use both /* and //. I propose you change to //. Also, if doing then a space between // and the text is nicer.